src/stream.c

```diff
@@ -295,8 +295,15 @@ static void rtp_handler(const struct sa *src, const struct rtp_header *hdr,
 		flush = true;
 	}
 
-	if (s->jbuf) {
+	/* payload-type changed? */
+	if (s->pt_dec != hdr->pt) {
```

Comment:
pt = 0 is also a valid payload type. If the first incoming packet is pt=0, this logic will not work.

s->pt_dec should be set to -1, similar to audio.c